x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -234,6 +234,19 @@ private void checkRequiredFields(Set<String> fields) {
         }
     }
 
+    private void checkFieldsWithCardinalityLimit() {
+        for (Map.Entry<String, Long> entry : config.getAnalysis().getFieldCardinalityLimits().entrySet()) {
+            String fieldName = entry.getKey();
+            long limit = entry.getValue();
+            long cardinality = fieldCardinalities.get(fieldName);
```

Comment:
Could this line cause NPE if `fieldName` is not in the index?

Reply:
No, we check this when we run the aggregation. This part of the code should assume there will be a limit for each field that should have one. If not it's a programmatic error.